src/snowflake/cli/plugins/nativeapp/teardown_processor.py

```diff
@@ -89,9 +112,60 @@ def drop_application(self, auto_yes: bool):
                     cc.message(f"Did not drop application object {self.app_name}.")
                     return  # The user desires to keep the app, therefore exit gracefully
 
-        # 4. All validations have passed, drop object
+        # 4. Check for application objects owned by the application
+        application_objects = self.get_application_objects()
+        if len(application_objects) > 0:
+            application_objects_str = self._application_objects_to_str(
+                application_objects
+            )
+            if cascade is True:
+                cc.message(
+                    dedent(
+                        f"""\
+                            The following application objects will be deleted:
+                            {application_objects_str}
+                        """
+                    )
+                )
+            elif cascade is False:
+                cc.message(
+                    dedent(
+                        f"""\
+                            The following application objects will not be deleted:
+                            {application_objects_str}
+                        """
+                    )
+                )
+            elif interactive:
+                if interactive:
+                    cascade = typer.confirm(
+                        dedent(
+                            f"""\
+                        The following application objects are owned by this application:
+                        {application_objects_str}
+                        Do you confirm to delete these?
+                    """
+                        )
+                    )
+            else:
+                raise ClickException(
+                    dedent(
+                        f"""\
+                    The following application objects are owned by this application:
+                    {application_objects_str}
+                    Please explicitly set --cascade if they should be deleted, or transfer ownership and run teardown again.
+                """
+                    )
+                )
+        elif cascade is None:
+            cascade = False
```

Comment:
Is this reachable? If so, how?

Reply:
Yes, if cascade is unset and there are no objects owned by the application. In this case I prefer the value to be explicitly `False` instead of `None`